test/foundry/deploy/17_holographer_tests.t.sol

```diff
@@ -0,0 +1,248 @@
+// SPDX-License-Identifier: UNLICENSED
+pragma solidity 0.8.13;
+
+import {Test, Vm, console} from "forge-std/Test.sol";
+import {Constants, ErrorConstants} from "../utils/Constants.sol";
+import {Holographer} from "../../../src/enforcer/Holographer.sol";
+import {Holograph} from "../../../src/Holograph.sol";
+import {MockExternalCall} from "../../../src/mock/MockExternalCall.sol";
+import {SampleERC721} from "../../../src/token/SampleERC721.sol";
+import {HelperDeploymentConfig} from "../utils/HelperDeploymentConfig.sol";
+import {DeploymentConfig} from "../../../src/struct/DeploymentConfig.sol";
+import {HelperSignEthMessage} from "../utils/HelperSignEthMessage.sol";
+import {Verification} from "../../../src/struct/Verification.sol";
+import {HolographFactory} from "../../../src/HolographFactory.sol";
+
+/**
+ * @title Testing the Holographer
+ * @notice Suite of unit tests for the Holographer contract
+ * @dev Translation of a suite of Hardhat tests found in test/17_holographer_tests.ts
+ */
+contract HolographerTests is Test {
+  MockExternalCall mockExternalCall;
+  HolographFactory factory;
+  SampleERC721 sampleERC721;
+  Holograph holograph;
+  Holographer holographer;
+  address zeroAddress = Constants.getZeroAddress();
+  uint256 privateKeyDeployer = Constants.getPKDeployer();
+  address deployer = vm.addr(privateKeyDeployer);
+  uint256 localHostFork;
+  string LOCALHOST_RPC_URL = vm.envString("LOCALHOST_RPC_URL");
+
+  function generateInitCode() public pure returns (bytes memory) {
```

Comment:
this function is the same that getInitCodeSampleErc721
If not like the name erc721, could be refactor that generic name

Reply:
Done